### PR TITLE
refactor(Core): fix some warnings from VS

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  Fix some warnings that I found in VS

```
Future.cpp -> warning C6011: dereferencing NULL pointer 'this->value_'
World.h -> warning C26451: Arithmetic overflow: Using operator '*' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '*' to avoid overflow
Channel.h -> warning C26451: Arithmetic overflow: Using operator '+' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '+' to avoid overflow
Field.h -> warning C26451: Arithmetic overflow: Using operator '+' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '+' to avoid overflow
Format.h -> warning C26451: Arithmetic overflow: Using operator '*' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '*' to avoid overflow
Hash_Map_Manager_T.cpp -> warning C6011: dereferencing NULL pointer 'entry'
```

## Tests Performed:
- Builds

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: deps/acelite/ace/Hash_Map_Manager_T.cpp [hunk 1]
```diff
@@ -239,14 +239,18 @@ ACE_Hash_Map_Manager_Ex<EXT_ID, INT_ID, HASH_KEY, COMPARE_KEYS, ACE_LOCK>::bind_
...
+          entry = new (ptr) ACE_Hash_Map_Entry<EXT_ID, INT_ID>(ext_id,
+              int_id,
+              this->table_[loc].next_,
+              &this->table_[loc]);
+
+          this->table_[loc].next_ = entry;
+          entry->next_->prev_ = entry;
+          ++this->cur_size_;
+          return 0;
+      }
```
Comment (Kitzunu) [inline deps/acelite/ace/Hash_Map_Manager_T.cpp:253]: I am not sure if the if statement is placed in the correct place here?

**Comment by Kitzunu** [comment]
Actually I dont know how to fix the docker error
```
fatal error: comparison of integers of different signs: 'unsigned long' and 'const time_t' (aka 'const long')
```
First of all I changed lastSpeakTime to unit64 so shouldn't that be unsigned long long and uint32 (previous) unsigned long? So why did it work before and not now? O.O

File: src/server/game/Chat/Channels/Channel.h [hunk 2]
```diff
@@ -164,9 +164,9 @@ class Channel
             if (state) flags |= MEMBER_FLAG_MUTED;
             else flags &= ~MEMBER_FLAG_MUTED;
         }
-        bool IsAllowedToSpeak(uint32 speakDelay) // pussywizard
+        bool IsAllowedToSpeak(uint64 speakDelay) // pussywizard
         {
-            if (lastSpeakTime + speakDelay <= sWorld->GetGameTime())
+            if (lastSpeakTime + speakDelay <= uint64(sWorld->GetGameTime()))
```
Comment (Kitzunu) [inline src/server/game/Chat/Channels/Channel.h:169]: Okay I assume this is hacky as fuck to make the build work for lastSpeakTime xD (maybe not, I don't know? I am no pro at this)

If this is hack af, how the fuck could I fix this to work xD

Comment (FrancescoBorzi) [reply src/server/game/Chat/Channels/Channel.h:169]: what about `static_cast<uint64>` instead?

Comment (Kitzunu) [reply src/server/game/Chat/Channels/Channel.h:169]: I don't know what that does fully, but I read a bit on it. but then you mean
```suggestion
        bool IsAllowedToSpeak(static_cast<uint64>(speakDelay)) // pussywizard
        {
                    if (lastSpeakTime + speakDelay <= static_cast<uint64>(sWorld->GetGameTime()))
```

Comment (FrancescoBorzi) [reply src/server/game/Chat/Channels/Channel.h:169]: yes

File: deps/acelite/ace/Future.cpp [hunk 3]
```diff
@@ -230,8 +230,12 @@ ACE_Future_Rep<T>::get (T &value,
       // Destructor releases the lock.
     }
 
-  value = *this->value_;
-  return 0;
+  if (value)
```
Comment (Yehonal) [inline deps/acelite/ace/Future.cpp:233]: Uhm, fixing/changing something in a dependency is something that we should not do if not properly tracked.

File: deps/acelite/ace/Hash_Map_Manager_T.cpp [hunk 4]
```diff
@@ -239,14 +239,18 @@ ACE_Hash_Map_Manager_Ex<EXT_ID, INT_ID, HASH_KEY, COMPARE_KEYS, ACE_LOCK>::bind_
...
 
-      entry = new (ptr) ACE_Hash_Map_Entry<EXT_ID, INT_ID> (ext_id,
-                                                            int_id,
-                                                            this->table_[loc].next_,
-                                                            &this->table_[loc]);
-      this->table_[loc].next_ = entry;
-      entry->next_->prev_ = entry;
-      ++this->cur_size_;
-      return 0;
+      if (entry)
```
Comment (Yehonal) [inline deps/acelite/ace/Hash_Map_Manager_T.cpp:242]: Same above

File: deps/fmt/fmt/format.h [hunk 5]
```diff
@@ -732,7 +732,7 @@ class decimal_formatter {
  private:
   char* buffer_;
 
-  void write_pair(unsigned N, uint32_t index) {
+  void write_pair(unsigned N, uint64_t index) {
```
Comment (Yehonal) [inline deps/fmt/fmt/format.h:735]: Same above

File: src/server/game/Chat/Channels/Channel.h [hunk 6]
```diff
@@ -164,9 +164,9 @@ class Channel
             if (state) flags |= MEMBER_FLAG_MUTED;
             else flags &= ~MEMBER_FLAG_MUTED;
         }
-        bool IsAllowedToSpeak(uint32 speakDelay) // pussywizard
+        bool IsAllowedToSpeak(static_cast<uint64>(speakDelay)) // pussywizard
```
Comment (FrancescoBorzi) [inline src/server/game/Chat/Channels/Channel.h:167]: this line looks weird to me, shouldn't it be just `bool IsAllowedToSpeak(uint64 speakDelay)` ? without any type of cast

the rest looks good to me

Comment (Kitzunu) [reply src/server/game/Chat/Channels/Channel.h:167]: I thought you meant I should change that oo Lol

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by pklloveyou** [comment]
Tested

**Comment by pklloveyou** [comment]
Can I add it here
SpellInfo.cpp
```
--        switch (PowerType)
++       switch ((Powers)PowerType)
```

**Pull request merged**